org/mozilla/jss/util/NativeProxy.java

```diff
@@ -31,20 +31,35 @@
 
     /**
      * Create a NativeProxy from a byte array representing a C pointer.
-     * This is the only way to create a NativeProxy, it should be called
+     * This is the primary way of creating a NativeProxy; it should be called
      * from the constructor of your subclass.
      *
      * @param pointer A byte array, created with JSS_ptrToByteArray, that
      * contains a pointer pointing to a native data structure.  The
      * NativeProxy instance acts as a proxy for that native data structure.
      */
     public NativeProxy(byte[] pointer) {
-        assert(pointer!=null);
+        this(pointer, true);
+    }
+
+    /**
+     * Create a NativeProxy from a byte array representing a C pointer.
+     * This allows for creating an untracked NativeProxy instance (when
+     * track=false), which allows for creating NativeProxy instances out
+     * of stack-allocated variables and/or creating NativeProxies which
+     * aren't freed.
+     */
+    public NativeProxy(byte[] pointer, boolean track) {
```

Comment:
@edewata  Should I make this protected to only allow its use by `StaticVoidPointer` and `NativeProxy`?

Reply:
It's really up to you. Basically if you're not really sure the method will be safe or useful for general usage then you can set it to protected, at least for now. If you're pretty sure it's safe/useful then it can be public. And we can always change or deprecate it later.

Reply:
There are other ways too. Instead of providing an option (i.e. `track` parameter) in `NativeProxy`, you could put the code in a method (e.g. `track()`) then override it in the subclass to change its behavior for that particular class.

It's also possible to check which class is executing the code (sounds familiar?) and then run the appropriate code. This is not recommended though. :)

Reply:
I've changed it to protected.